packages/graphql-generator/src/statements.ts

```diff
@@ -0,0 +1,52 @@
+const { GraphQLStatementsFormatter } = require('./utils');
+import { generateGraphQLDocuments } from '@aws-amplify/graphql-docs-generator';
+import type { GeneratedOperations } from '@aws-amplify/graphql-docs-generator';
+import { GenerateStatementsOptions, StatementsTarget, GeneratedOutput } from './typescript';
+import { statementsTargetToFileExtensionMap } from './utils';
+
+export function generateStatements(options: GenerateStatementsOptions): GeneratedOutput {
+  const { schema, target, maxDepth = 2, typenameIntrospection = true } = options;
+
+  if (!Object.keys(statementsTargetToFileExtensionMap).includes(target)) {
+    throw new Error(`${target} is not supported a supported target.`);
+  }
+
+  const generatedOperations = generateGraphQLDocuments(schema, {
+    maxDepth,
+    useExternalFragmentForS3Object: target === 'graphql',
+    typenameIntrospection,
+  });
+  return generatedDocuments(target, generatedOperations);
+}
+
+function generatedDocuments(target: StatementsTarget, generatedStatements: GeneratedOperations): GeneratedOutput {
```

Comment:
Seems like this should be named `generateDocuments` to align w/ the verbiage on `generateStatements`.